core/object/callable_method_pointer.cpp

```diff
@@ -38,13 +38,7 @@ bool CallableCustomMethodPointerBase::compare_equal(const CallableCustom *p_a, c
 		return false;
 	}
 
-	for (uint32_t i = 0; i < a->comp_size; i++) {
-		if (a->comp_ptr[i] != b->comp_ptr[i]) {
-			return false;
-		}
-	}
-
-	return true;
+	return memcmp(a->comp_ptr, b->comp_ptr, a->comp_size * 4) == 0;
```

Comment:
```suggestion
	// Avoid hiving by memory address proximity, which leads to unpredictable performance over time
	// due to the reuse of old addresses for newer objects. Use byte-wise comparison to leverage backwords
	// encoding as a way to decouple spatiality and time.
	return memcmp(a->comp_ptr, b->comp_ptr, a->comp_size * 4) == 0;
```

Feel free to reword the comment. I tried to paraphrase your rationale to also check that I understand it.

Reply:
> Feel free to reword the comment. I tried to paraphrase your rationale to also check that I understand it.

:ok_hand: 

This has the dubious honor of being the most wtf bug I've ever come across